Add README.md

Zer0Legion opened the pull request:
# Squid 

## Overview
Introducing **Squid**! No, he's not seafood 🦑. Instead, he's a black cat! 🐈‍⬛ 
> I'll handle your tasks for you.. Just feed me after. - Squid

### He can:

- Keep track of your tasks
- Mark (and unmark!) tasks as completed!
- Delete tasks!

### He supports:

1. **Todo** Tasks (_No deadline_)
2. **Deadline** Tasks (_1 deadline_)
3. **Event** Tasks (_1 from and 1 to date_)

### Compensating him for his efforts is as simple as this!
```Java
Squid.feed();
```

### Upcoming features:
- [ ] `help` command

Created by Leng Khai https://github.com/Zer0Legion

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Zer0Legion committed:
ignore binary/temp files

Zer0Legion committed:
Rename, greet

Zer0Legion committed:
Implement exit message

Zer0Legion committed:
update .gitignore

Zer0Legion committed:
Implement echo; update welcome and bye message

Zer0Legion committed:
implement add/list with ArrayList<String>

Zer0Legion committed:
Implement Task class

Zer0Legion committed:
Implemented mark and unmark

Zer0Legion committed:
half implemented todo.

Zer0Legion committed:
added todo

Zer0Legion committed:
Added support for todo; Task.toString() method

Zer0Legion committed:
added todo, event and deadline. As for previous add function, assumed it to be a todo type because no date was specified. Still keep previous behaviour of default adding a task when no command was specified.
Also moved all the string messages to a common dictionary.

Zer0Legion committed:
added string formatting for deadline and event

Zer0Legion committed:
Implemented inheritance; turned Task abstract, with 3 subclasses Todo, Event, Deadline

Zer0Legion committed:
Write test cases for:
add
list
todo
event
deadline
mark
unmark
bye

Zer0Legion committed:
add testcase for echo

Zer0Legion committed:
Handle errors for empty description and unknown command

Zer0Legion committed:
Handled: duplicate task name
unknown command
wrrong arguments

Zer0Legion committed:
started to implement delete

Zer0Legion committed:
add on to delete

Zer0Legion committed:
fully implement delete task

Zer0Legion committed:
Merge commit '9ab14cedeb3b92debbaee6960d0ad2c07acb8af8'

# Conflicts:
#	src/main/java/Duke.java

Zer0Legion committed:
Extracted string constants into package

Zer0Legion committed:
Extract classes; exceptions into packages etc.

Zer0Legion committed:
Start to implement Storage

Zer0Legion committed:
implement string formatting for stored tasks

Zer0Legion committed:
ignore hard disk storage file

Zer0Legion committed:
ignore hard disk storage file

Zer0Legion committed:
Implement save

Zer0Legion committed:
Merge branch 'branch-Level-7' of https://github.com/Zer0Legion/ip into branch-Level-7

# Conflicts:
#	.gitignore

Zer0Legion committed:
Added support for 100 date and 6 time formats; also fixed minor bug in mark/unmark

Zer0Legion committed:
Refactoring different methods for formatting date and time

Zer0Legion committed:
Extract the parsing logic of date and time into their own methods

Zer0Legion committed:
package all files into main squid package

Zer0Legion committed:
delete unused variable

Zer0Legion committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Zer0Legion committed:
Refactor: rename packages to correct casing

Zer0Legion committed:
Refactor: add more patterns for date

Zer0Legion committed:
Refactor: Extract constants

Zer0Legion committed:
Build: include build.gradle

Zer0Legion committed:
Refactor: move test file to correct spot

Zer0Legion committed:
Fix: update correct import

Zer0Legion committed:
Testing: create new TaskInterface

Zer0Legion committed:
Testing: Added 11 tests for 5 methods and 3 exception cases

Zer0Legion committed:
Refactor: delete unused class

Zer0Legion committed:
Testing: add tests for 4 methods

Zer0Legion committed:
Testing: add tests for all public methods

Zer0Legion committed:
Build: create jar file

Zer0Legion committed:
Add: find feature

Zer0Legion committed:
Style: add JavaDoc

Zer0Legion committed:
Style: minimal changes to coding standard

Zer0Legion committed:
Merge branch 'branch-A-CodingStandard'

Zer0Legion committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/squid/Duke.java
#	src/main/java/squid/tasks/Tasks.java

Zer0Legion committed:
Style: use AB3 checkstyle

Zer0Legion committed:
Style: fixed accordingly

Zer0Legion committed:
Merge branch 'branch-A-CheckStyle'

Zer0Legion committed:
Code: add increments for ui

Zer0Legion committed:
Refactor: modify method signatures to return String

Zer0Legion committed:
Feat: add iteration 3 of javafx

Zer0Legion committed:
feat: add iteration 3 of javafx

Zer0Legion committed:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

Zer0Legion committed:
Feat: add MainWindow.java

Zer0Legion committed:
Feat: add Main.java

Zer0Legion committed:
Feat: integrate stagebuilder

Zer0Legion committed:
Feat: commented out unused code

soons1 commented on src/main/java/squid/Squid.java:
```diff
@@ -0,0 +1,353 @@
+        hello();
+
+        Scanner scanner = new Scanner(System.in);
+        boolean loop = true;
```
Would it be better to name the loop variable to sound like a boolean (eg. isLoop)?

soons1 commented on src/test/java/squid/tasks/TasksTest.java:
```diff
@@ -0,0 +1,166 @@
+
+    @Test
+    public void testParseTask_taskCompleted() {
+        boolean expected = false;
```
would isExpected be a better variable name instead?

soons1 commented on src/main/java/squid/tasks/Task.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public abstract class Task implements TaskInterface {
+
+    protected boolean completed;
```
Would it be better to name the variable isCompleted instead for clarity?

soons1 commented on src/main/java/squid/Squid.java:
```diff
@@ -0,0 +1,353 @@
+        if (params.length <= 1) {
+            throw new NotEnoughInputsException(
+                    String.format(
+                            Exceptions.NOT_ENOUGH_INPUTS, "echo", CorrectUsage.ECHO));
```
Is there a reason why line 85 is separated from line 84 as compared to line 264?

soons1 commented on src/main/java/squid/Squid.java:
```diff
@@ -0,0 +1,353 @@
+        if (params.length <= 1) {
+            throw new NotEnoughInputsException(
+                    String.format(
+                            Exceptions.NOT_ENOUGH_INPUTS, "todo", CorrectUsage.TODO));
```
Would it be better to keep line 110 and 111 as one line for consistency?

soons1 reviewed (commented):
LGTM 👍 
The issues I have pointed out are very minor and up for discussion!

Zer0Legion committed:
Feat: add working echo ui

Zer0Legion committed:
Feat: fully implement ui

Zer0Legion committed:
Feat: fully implement ui

Zer0Legion committed:
Merge branch 'Branch-Ui-Redo' of https://github.com/Zer0Legion/ip into Branch-Ui-Redo

# Conflicts:
#	src/main/java/squid/Squid.java

Zer0Legion committed:
Fix: minor issue with blank input

Zer0Legion committed:
Feat: save every input

Zer0Legion committed:
Merge branch 'branch-Level-10' into Branch-Ui-Redo

# Conflicts:
#	src/main/java/squid/Squid.java
#	src/main/java/squid/ui/DialogBox.java

Zer0Legion committed:
Merge branch 'Branch-Ui-Redo'

Zer0Legion committed:
Fix: checkstyle

Zer0Legion committed:
Build: change main directory

Zer0Legion committed:
Fix: minor UI bug

Zer0Legion committed:
Feat: add assertions

There are methods that require a a file in the file path to exist.

Checks need to be done to ensure the file location already exists to
prevent IO Exceptions.

Using assertions allows us to assume that file location already
exists at the point of saving or reading.

Let's assert that the file location exists during IO operations.

Zer0Legion committed:
Revert "Feat: add assertions"

This reverts commit b8a9024ddcfbbc8616961b468c93cc855ca63142.

Zer0Legion committed:
Feat: add assertions

There are methods that require a a file in the file path to exist.

Checks need to be done to ensure the file location already exists to
prevent IO Exceptions.

Using assertions allows us to assume that file location already
exists at the point of saving or reading.

Let's assert that the file location exists during IO operations.

Zer0Legion committed:
Merge branch 'branch-A-Assertions'

Zer0Legion committed:
Reverse commit

Zer0Legion committed:
Feat: add assertions

There are methods that require a a file in the file path to exist.

Checks need to be done to ensure the file location already exists to
prevent IO Exceptions.

Using assertions allows us to assume that file location already
exists at the point of saving or reading.

Let's assert that the file location exists during IO operations.

Zer0Legion committed:
Merge pull request #1 from Zer0Legion/branch-A-Assertions

Feat: add assertions

Zer0Legion committed:
Merge branch 'branch-A-Assertions'

Zer0Legion committed:
Merge remote-tracking branch 'origin/master'

Zer0Legion committed:
Build: add CI workflow

Zer0Legion committed:
Merge pull request #2 from Zer0Legion/branch-A-CI

Build: add CI workflow

Zer0Legion committed:
Feat: add help command

Zer0Legion committed:
Build: move gradle.yml

Zer0Legion committed:
Style: checkstyle

Zer0Legion committed:
Style: checkstyle

Zer0Legion committed:
Merge pull request #3 from Zer0Legion/branch-C-Help

Feat: add help command

Zer0Legion committed:
Build: move gradle.yml

Zer0Legion committed:
Merge pull request #4 from Zer0Legion/branch-C-Help

Build: move gradle.yml

Zer0Legion committed:
Build: edit gradle.yml

shavonneg commented on src/main/java/squid/tasks/Event.java:
```diff
@@ -0,0 +1,67 @@
+import squid.constants.Messages;
+import squid.constants.Regex;
+
+/**
+ * Class encapsulating Deadline tasks.
+ */
```
I think you meant Event class here

shavonneg commented on src/main/java/squid/tasks/Task.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * @return Whether the task is complete.
+     */
+    public boolean isCompleted() {
```
Nice that you followed the coding standard to make the method sound like booleans.

shavonneg commented on src/main/java/squid/tasks/Tasks.java:
```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    /**
+     * Reads current tasks from hard disk storage.
+     * @throws ParseFailException If a string is unable to be parsed.
+     * @throws DuplicateTaskNameException If there are duplicate task names in the storage file.
+     * @throws SquidDateException If a DateTime object is unable to be parsed.
+     */
```
I like the clarity here for the javadocs, and it's nice that it's properly aligned too!

shavonneg commented on src/main/java/squid/constants/Exceptions.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * The class of constants for Strings regarding exception messages.
+ */
+public class Exceptions {
+    public static final String INCORRECT_INPUT = "You haven't taught me that trick yet..";
+    public static final String NOT_ENOUGH_INPUTS = "hmm, %s what?\nUsage: %s";
+    public static final String NOT_ENOUGH_DATES =
+            "You need a name for the task and a total of %d date(s) for a %s task!\nUsage: %s";
+    public static final String DUPLICATE_TASK_NAME = "There already exists a task with the name %s!";
+    public static final String INCORRECT_INDEX = "Please enter a valid integer!\nUsage: %s";
+    public static final String PARSE_FAIL = "Not parsing a Task String: \n%s";
+    public static final String SQUID_DATE =
+            "I can't read this! \n%s\nInput the %s in a cat-readable format, something like:\n%s";
```
I like that you printed out strings to show users clearly what the error is that they encounter.

shavonneg reviewed (commented):
LGTM, just nitpicking on minor things!

Zer0Legion committed:
refactor: rename all instances of duke to squid

Zer0Legion committed:
docs: add screenshot

Zer0Legion committed:
docs: update README.md

Zer0Legion committed:
fix: minor typo

Zer0Legion committed:
build: fix version

Zer0Legion committed:
docs: update image

Zer0Legion committed:
fix: find command

Zer0Legion committed:
style: fix checkstyle

Zer0Legion committed:
style: fix checkstyle

Zer0Legion committed:
ui defaults

Zer0Legion committed:
Fix ui clipping

Zer0Legion committed:
Fix find command

Zer0Legion committed:
Update help message

Zer0Legion committed:
Fix checkstyle

Zer0Legion committed:
Merge pull request #7 from Zer0Legion/branch-fix-ui

Branch fix UI
